src/System.Private.CoreLib/src/System/String.cs

```diff
@@ -2743,12 +2749,34 @@ public String Replace(char oldChar, char newChar)
                 {
                     fixed (char* pResult = &result._firstChar)
                     {
-                        //Copy the characters, doing the replacement as we go.
-                        for (int i = 0; i < firstFoundIndex; i++)
-                            pResult[i] = pChars[i];
+                        pSrc = pChars;
+                        pDst = pResult;
+                        int copyLength = Length;
 
-                        for (int i = firstFoundIndex; i < Length; i++)
-                            pResult[i] = (pChars[i] == oldChar) ? newChar : pChars[i];
+                        //Copy the characters already proven not to match.
+                        while (copyLength > remainingLength)
```

Comment:
The loss using wstrcpy is very minimal, about 2ns for 10 chars, still a fair way ahead. Large benefit for longer strings 663ns goes to 493ns.